File: Longest-String/Longest-string.rb
def length_of_longest_substring(s)
   array = s.split("")
   checker =[];
   longest_length = 0
  (0..array.length - 1).each do |i|
    answer = checker.include(s[i])
     if answer 
        checker =[];
        longest_length = 0
      
     else 
        checker.push(s[i])
        longest_length ++
     end

  end
  return longest_length
end

length_of_longest_substring("Donat")

File: Longest-String/Longest-string.js
const Longest = (s) => {
  let checker = [];
  let length_array = [];
  let length_counter = 0;
  let string = s.split("");
  for (let i = 0; i < string.length; i++) {
    let include = checker.include?(string[i]);
    if (include) {
      checker = [];
      length_counter = 0
    }
    else {
      checker.push(string[i]);
      length_counter = length_counter + 1;
      length_array.push(length_counter);
    }
  }

  console.log(s);
};
